packages/core/src/navigation/navigation-group.element.ts

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (c) 2016-2021 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { html, internalProperty, LitElement, PropertyValues } from 'lit-element';
+
+import {
+  Animatable,
+  animate,
+  AnimationNavigationGroupOpenName,
+  baseStyles,
+  createId,
+  event,
+  EventEmitter,
+  i18n,
+  I18nService,
+  id,
+  property,
+  querySlot,
+  querySlotAll,
+  reverseAnimation,
+  syncProps,
+} from '@cds/core/internal';
+import { styles } from './navigation-group.element.css.js';
+import { NavigationLayout } from './interfaces/navigation-layout.interface.js';
+import { DEFAULT_NAVIGATION_LAYOUT } from './utils/index.js';
+import { CdsNavigationItem } from './navigation-item.element';
+import { CdsNavigationStart } from './navigation-start.element';
+
+export const CdsNavigationGroupTagName = 'cds-navigation-group';
+
+/**
+ *
+ * ```typescript
+ * import '@cds/core/navigation/register.js';
+ * ```
+ *
+ * ```html
+ * <cds-navigation-group>
+ *
+ * </cds-navigation-group>
+ * ```
+ *
+ * @element cds-navigation-group
+ * @slot
+ */
+@animate({
+  expanded: {
+    true: AnimationNavigationGroupOpenName,
+    false: reverseAnimation(AnimationNavigationGroupOpenName),
+  },
+})
+export class CdsNavigationGroup extends LitElement implements Animatable {
+  @property({ type: String })
+  cdsMotion = 'on';
+
+  @event()
+  cdsMotionChange: EventEmitter<string>;
+
+  @event()
+  expandedChange: EventEmitter<boolean>;
+
+  /**
+   * @desc
+   * Associate the cds-navigation-button in this template with projected into the
+   * aria-labelledby wrapper region.
+   *
+   * @private
+   */
+  @id() navigationGroupId: string;
+
+  @i18n() i18n = I18nService.keys.navigation;
+
+  /**
+   * @description
+   * Getter method for a reference to the selector cds-navigation-group > cds-navigation-start
+   * This lets each group flag its cds-navigation-start element and sync that info down. This is
+   * needed because cds-navigation-start elements can be used at the root level and inside
+   * cds-navigation-group elements.
+   *
+   * @private
+   */
+  @internalProperty()
+  private get isGroupStart(): boolean {
+    return !!this.groupStart;
+  }
+
+  /**
+   * @desc
+   * The element id is used to update cds-navigation aria-activedescendent when a group is the current
+   * active item.
+   */
+  @property({ type: String, reflect: true, attribute: 'id' })
+  id: string = createId();
+
+  @internalProperty()
```

Comment:
I defined a type for this and hasFocus because it is similar. How for do you think I should go with this? There are a bunch of boolean properties that could follow the same pattern. Is there something in the dev guidlines I missed that talks about this?